src/hugr/serialize.rs

```diff
@@ -114,23 +123,31 @@ impl TryFrom<&Hugr> for SerHugrV0 {
             *parent = node_rekey[parent];
         }
 
-        let find_offset = |p: SubportIndex| {
-            (
-                node_rekey[&graph.port_node(p).unwrap()],
-                graph.port_offset(p).unwrap().index(),
-            )
+        let find_offset = |node: NodeIndex, offset: usize, dir: Direction| {
+            let sig = &op_types[node].signature();
+            let offset = match offset < sig.port_count(dir) {
+                true => Some(offset as u16),
```

Comment:
This cast worries me a bit -- is there some inconsistency in integer types that we should iron out?

Reply:
Portgraph supports up to `u16::MAX` ports per node in each direction for performance reasons. Any PortOffset higher than 65k would panic during runtime.

(Note that that limit is _per node_, the graph itself can support up to `usize`)